census_view.cpp

```diff
@@ -963,7 +1929,7 @@ BEGIN_EVENT_TABLE(CensusView, ViewEx)
     EVT_UPDATE_UI(XRCID("copy_census"          ),CensusView::UponUpdateColumnValuesVary )
     EVT_UPDATE_UI(XRCID("paste_census"         ),CensusView::UponUpdateAlwaysEnabled    )
     EVT_UPDATE_UI(XRCID("add_cell"             ),CensusView::UponUpdateAlwaysEnabled    )
-    EVT_UPDATE_UI(XRCID("delete_cells"         ),CensusView::UponUpdateNonemptySelection)
+    EVT_UPDATE_UI(XRCID("delete_cells"         ),CensusView::UponUpdateAlwaysEnabled    )
```

Comment:
This looks wrong to me, it changes the behaviour of the existing DVC-based version. Am I missing something or should this change be reverted?

Reply:
When we press an arrow key, we loose the selection. But we still have the current cell, which index will be used as the row to be deleted.

See `UponDeleteCells(wxCommandEvent&)`:
```cpp
    auto rows = grid_window_->GetSelectedRows();
    // If no cell is selected then use the current row
    if (rows.empty())
        rows.push_back(grid_window_->GetGridCursorRow());
```

> `UponUpdateNonemptySelection()` is not used at all currently, which is definitely wrong

Agreed.

Reply:
The problem is not with the grid version, but with the DVC one, which checked the selection size before but doesn't do it any longer. It looks like the selection is never empty for DVC neither (which surprised me, because it definitely _looks_ like it is empty), but I nevertheless prefer not to change the existing code, so I kept `UponUpdateNonemptySelection()` in `CensusDVCView` only.